lrpar/src/lib/ctbuilder.rs

```diff
@@ -79,14 +79,71 @@ pub enum ActionKind {
     GenericParseTree
 }
 
+struct CTConflictsError<StorageT: Eq + Hash> {
+    pub grm: YaccGrammar<StorageT>,
+    pub sgraph: StateGraph<StorageT>,
+    pub stable: StateTable<StorageT>
+}
+
+impl<StorageT> fmt::Display for CTConflictsError<StorageT>
+where
+    StorageT: 'static + Debug + Hash + PrimInt + Serialize + TypeName + Unsigned,
+    usize: AsPrimitive<StorageT>,
+    u32: AsPrimitive<StorageT>
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let conflicts = self.stable.conflicts().unwrap();
+        write!(
+            f,
+            "CTConflictsError{{{} Shift/Reduce, {} Reduce/Reduce}}",
+            conflicts.sr_len(),
+            conflicts.rr_len()
+        )
+    }
+}
+
+impl<StorageT> fmt::Debug for CTConflictsError<StorageT>
+where
+    StorageT: 'static + Debug + Hash + PrimInt + Serialize + TypeName + Unsigned,
+    usize: AsPrimitive<StorageT>,
+    u32: AsPrimitive<StorageT>
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let conflicts = self.stable.conflicts().unwrap();
+        write!(
+            f,
+            "CTConflictsError{{{} Shift/Reduce, {} Reduce/Reduce}}",
+            conflicts.sr_len(),
+            conflicts.rr_len()
+        )
+    }
+}
+
+impl<StorageT> Error for CTConflictsError<StorageT>
+where
+    StorageT: 'static + Debug + Hash + PrimInt + Serialize + TypeName + Unsigned,
+    usize: AsPrimitive<StorageT>,
+    u32: AsPrimitive<StorageT>
+{
+}
+
 /// A `CTParserBuilder` allows one to specify the criteria for building a statically generated
 /// parser.
-pub struct CTParserBuilder<StorageT = u32> {
+pub struct CTParserBuilder<StorageT = u32>
+where
+    StorageT: Eq + Hash
+{
     // Anything stored in here almost certainly needs to be included as part of the rebuild_cache
     // function below so that, if it's changed, the grammar is rebuilt.
     recoverer: RecoveryKind,
     phantom: PhantomData<StorageT>,
-    actionkind: ActionKind
+    actionkind: ActionKind,
+    error_on_conflicts: bool,
+    cdata: Option<(
```

Comment:
`cdata` is kind of a weird name: it certainly needs to be explicitly documented what it's for, and pointed out that it's not part of the cache. You'll need to update that comment above too slightly.

Reply:
Do you have an idea for a better name? Maybe just `data`?

Reply:
In a later comment, I sort-of suggest renaming this to `conflicts` (for other reasons).